### ISSUE TITLE
can't load custom fonts

### ISSUE THREAD

**Issue by radio-alice** [jaames/zfont#3]
local fonts don't load with any fields besides font.data

```
let font = new Zdog.Font({
  src: 'https://cdn.jsdelivr.net/gh/jaames/zfont/demo/fredokaone.ttf'
})
// works - font.htmx and font.hhea are present on loaded font

let afont = new Zdog.Font({
  src: './fonts/CSTMXprmntl03_Cyrillic.otf'
})
// doesn't work - only font.data present once loaded
```

trying to use a custom font results in: "Uncaught (in promise) TypeError: Cannot read property 'aWidth' of undefined", since there is no `htmx` field

This happens with every font including google fonts and open source fonts

**Comment by jaames** [comment]
I can't really say without being able to look at the font file myself. 

Maybe check that the font is being fetched correctly and that your server isn't responding to the font request with an error page or something. Also, have you had success with loading the fredokaone TTF from your local setup?

**Comment by jaames** [comment]
After looking into this a bit more, I'm fairly certain this is a case of a misconfigured server responding to your font requests with something *other* than a font file. Zfont was just naively assuming that the font request was successful if it returned with a HTTP status code between 200 and 300. 

I just pushed a quick fix (v 1.2.5) which adds a check for the presence of the font fields that Zfont requires (`hmtx`, `head`, `hhea`, etc) and throws a more insightful error message if they couldn't be found for whatever reason.

Hope that helps @radio-alice :) I'm going to close this issue now but feel free to reopen/comment/whatever if you're still having trouble.

Cheers!
